src/generator/boardGenerator.java

```diff
@@ -0,0 +1,222 @@
+package generator;
+
+import java.util.Random;
+
+public class boardGenerator {
+    private int[][] board;
+    private int size;
+
+    public boardGenerator(int size) {
+        size--;
+        if (size < 3) {
+            this.size = 3;
+        } else {
+            this.size = size;
+        }
+        board = new int[size][size];
+        generate();
+        while (checkReq())// hehe, yeah, generate 'til...
+            generate();
+        while (!setValues())
+            resetBoard();
+        show();
+        System.out.println(toString());
+    }
+
+    //this generator could generate board with two fields that don't have routes to one another
+
+    private void generate() {
+        Random rand = new Random();
+        int n;
+        for (int i = 0; i != size; ++i) {
+            for (int j = 0; j != size; ++j) {
+                if ((n = rand.nextInt(2)) == 0)
+                    n = rand.nextInt(2);
+                board[i][j] = 0 - n;
+            }
+        }
+
+        if (size > 9) {
+            int length;
+            for (int i = 0; i != size; ++i) {
+                length = 0;
+                for (int j = 0; j != size; ++j) {
+                    if (board[i][j] == -1) {
+                        ++length;
+                    } else {
+                        length = 0;
+                    }
+                    if (length > 9) {
+                        board[i][j] = 0;
+                        length = 0;
+                    }
+                }
+
+            }
+            for (int i = 0; i != size; ++i) {
+                length = 0;
+                for (int j = 0; j != size; ++j) {
+                    if (board[j][i] == -1) {
+                        ++length;
+                    } else {
+                        length = 0;
+                    }
+                    if (length > 9) {
+                        board[j][i] = 0;
+                        length = 0;
+                    }
+                }
+
+            }
+        }
+
+
+        boolean isChanged = true;
+        while (isChanged) {
+            isChanged = false;
+            for (int i = 0; i != size; ++i) {
+                for (int j = 0; j != size; ++j) {
+                    if (board[i][j] == -1) {
+                        if ((i == 0 || board[i - 1][j] != -1) && (i == size - 1 || board[i + 1][j] != -1)) {
+                            board[i][j] = 0;
+                            isChanged = true;
+                        }
+                    }
+                    if (board[i][j] == -1) {
+                        if ((j == 0 || board[i][j - 1] != -1) && (j == size - 1 || board[i][j + 1] != -1)) {
+                            board[i][j] = 0;
+                            isChanged = true;
+                        }
+                    }
+                }
+            }
+        }
+
+    }
+
+    private boolean checkReq() {
```

Comment:
Maybe can we use better name? This check only if the BLANK fields are on less than 3/4 of space, nothing more, so e.g. `checkBlankFieldsSpaceReq` or something better